### PR TITLE
Queue

### PR DESCRIPTION
**Pull request by welliam**

### CODE REVIEW

File: src/test_queue.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    """Test dequeue method of queue."""
+    from queue import Queue
+    assert Queue(values).dequeue() == result
+
+
+@pytest.mark.parametrize('values', SIZE_TABLE)
+def test_size(values):
+    """Test size method of queue."""
+    from queue import Queue
+    assert Queue(values).size() == len(values)
```
Comment (cewing) [inline src/test_queue.py]: This works to verify that the `size` attribute of your queue works initially.  What about after you dequeue a value?  Or enqueue a new value?  Or first one, then the other?  What if you empty a queue and then check the size?  

There are a number of edge cases that are not covered by this test.

File: src/test_queue.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+
+def test_enqueue():
+    """Test enqueue method of queue."""
+    from queue import Queue
+    q = Queue()
+    q.enqueue(1)
+    q.enqueue(2)
+    q.enqueue(3)
+    assert q.size() == 3
```
Comment (cewing) [inline src/test_queue.py]: Is it enough to know that the size is three?  What if the items are not actually being stored?  Can you prove that the `1` you put in there first is actually in there somewhere?

File: src/test_queue.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    """Test peek method of queue."""
+    from queue import Queue
+    q = Queue()
+    q.enqueue(1)
+    q.enqueue(2)
+    q.enqueue(3)
+    assert q.peek() == 1
+    assert q.size() == 3
+    q.dequeue()
+    assert q.peek() == 2
```
Comment (cewing) [inline src/test_queue.py]: This is a bit more interesting a test.  However, I would urge you to think more about breaking this kind of test into different tests.  There are three things you are testing here:

1.  The first value you add is seen by peek
2.  The size of the list is the expected size
3.  after dequeuing the first value, the next value peeked is the expected one.

That's actually three different tests, one of which is really handled by the tests on enqueueing above.

File: src/test_queue.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+
+def test_peek_error():
+    """Test that peek throws IndexError when queue is empty."""
+    from queue import Queue
+    with pytest.raises(IndexError):
+        Queue().peek()
```
Comment (cewing) [inline src/test_queue.py]: Glad to see you asserting what should happens when you do the wrong thing.  I would like to see a lot more of this.

File: src/queue.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+        try:
+            return self.dll.shift()
+        except IndexError:
+            raise IndexError("dequeue called on empty queue.")
+
+    def peek(self):
+        """Return the value at the tail of the queue without removing it."""
+        try:
+            return self.dll.tail.value
+        except AttributeError:
```
Comment (WillWeatherford) [inline src/queue.py]: per the assignment specifics, peek() should just return none when the queue is empty.

File: src/test_queue.py (hunk 4 above)
Comment (WillWeatherford) [inline src/test_queue.py]: Per the assignment specifics, peek() should just return None when the Queue is empty.
